plugins/linux-pipewire/camera-portal.c

```diff
@@ -378,61 +373,68 @@ static void camera_format_list(struct camera_device *dev, obs_property_t *prop)
 		obs_data_set_int(data, "height", resolution.height);
 
 		framerate_prop = spa_pod_find_prop(p->param, NULL, SPA_FORMAT_VIDEO_framerate);
-		if (!framerate_prop)
-			continue;
+		if (framerate_prop) {
+			struct spa_pod *framerate_pod;
+			uint32_t n_framerates;
+			enum spa_choice_type framerate_choice;
+			const struct spa_fraction *framerate_values;
+			g_autoptr(GArray) framerates = NULL;
+
+			framerate_pod = spa_pod_get_values(&framerate_prop->value, &n_framerates, &framerate_choice);
+			if (framerate_pod->type != SPA_TYPE_Fraction) {
+				blog(LOG_WARNING, "Framerate is not a fraction");
+				continue;
+			}
 
-		framerate_pod = spa_pod_get_values(&framerate_prop->value, &n_framerates, &framerate_choice);
-		if (framerate_pod->type != SPA_TYPE_Fraction) {
-			blog(LOG_WARNING, "Framerate is not a fraction");
-			continue;
-		}
+			framerate_values = SPA_POD_BODY(framerate_pod);
+			framerates = g_array_new(FALSE, FALSE, sizeof(struct spa_fraction));
 
-		framerate_values = SPA_POD_BODY(framerate_pod);
-		framerates = g_array_new(FALSE, FALSE, sizeof(struct spa_fraction));
+			switch (framerate_choice) {
+			case SPA_CHOICE_None:
+				g_array_append_val(framerates, framerate_values[0]);
+				break;
+			case SPA_CHOICE_Range:
+				blog(LOG_WARNING, "Ranged framerates not supported");
```

Comment:
See above - log messages, particularly public ones (`WARNING` and `ERROR`) need to carry enough context to be valuable to users attempting to self-help or supporters as our logging system is non-structured and cannot carry context information otherwise.

Reply:
Out of scope, those lines are marked as addition due to whitespace change